### PR TITLE
Titan: Add snapshot related test

### PR DESCRIPTION
**Pull request by Connor1996**
- add some logs
- add a test case: create a snapshot before flush, and test whether it is alright to get value after flush with that snapshot.

test passed after https://github.com/pingcap/rocksdb/pull/79 is merged.

### CODE REVIEW

File: utilities/titandb/db_impl.cc [hunk 1]
```diff
@@ -51,6 +51,8 @@ class TitanDBImpl::FileManager : public BlobFileManager {
       }
       if (!s.ok()) return s;
 
+      ROCKS_LOG_WARN(db_->db_options_.info_log, "Titan adding blob file [%llu]",
```
Comment (yiwu-arbug) [inline utilities/titandb/db_impl.cc]: Shall we go with info level? e.g. `ROCKS_LOG_INFO`

File: utilities/titandb/db_iter.h [hunk 2]
```diff
@@ -139,11 +139,11 @@ class TitanDBIterator : public Iterator {
...
       if (status_.IsCorruption()) {
-        fprintf(stderr, "key:%s GetBlobValue err:%s\n",
+        fprintf(stderr, "key:%s GetBlobValue err:%s with sequence number:%lu \n",
                 iter_->key().ToString(true).c_str(),
-                status_.ToString().c_str());
-        assert(false);
-        return false;
+                status_.ToString().c_str(),
+                options_.snapshot->GetSequenceNumber());
+        abort();
```
Comment (yiwu-arbug) [inline utilities/titandb/db_iter.h]: Instead of abort, shall we return proper status to caller (e.g. tikv) and let it handle the error?

Comment (yiwu-arbug) [reply utilities/titandb/db_iter.h]: The comment about `DeleteFilesInRanges` is not correct. Let's remove it.

Comment (Connor1996) [reply utilities/titandb/db_iter.h]: I think for inconsistency we'd better abort to reveal bugs.

Comment (yiwu-arbug) [reply utilities/titandb/db_iter.h]: I think we should return Status::Corruption() to tikv on read, and tikv can decide to abort.

Comment (Connor1996) [reply utilities/titandb/db_iter.h]: Okay, seems that TiKV didn't follow the best practice of iterator that check `status()` when `valid()` is false, I will export the `status()` in `rust-rocksdb`

File: utilities/titandb/db_iter.h [hunk 3]
```diff
@@ -134,16 +134,11 @@ class TitanDBIterator : public Iterator {
...
-      // cannot abort here.
       if (status_.IsCorruption()) {
-        fprintf(stderr, "key:%s GetBlobValue err:%s\n",
+        fprintf(stderr, "key:%s GetBlobValue err:%s with sequence number:%lu \n",
                 iter_->key().ToString(true).c_str(),
-                status_.ToString().c_str());
-        assert(false);
-        return false;
+                status_.ToString().c_str(),
+                options_.snapshot->GetSequenceNumber());
```
Comment (yiwu-arbug) [inline utilities/titandb/db_iter.h]: should return false as before.

but seems titan iterator don't return corrupted status correctly even if we return false here. We can fix it in another PR.

Comment (Connor1996) [reply utilities/titandb/db_iter.h]: L143 will check `status_` and return true which break the loop in `Next()` or `Prev()`, and then `Status()` will return the corrupted status as expected. Am I wrong？

Comment (yiwu-arbug) [reply utilities/titandb/db_iter.h]: sorry, I misread the code.

**Pull request merged**